File: Nx3CS/bin/test/src/nx3/xml/VoiceXML.cs

#pragma warning disable 109, 114, 219, 429, 168, 162
namespace nx3.xml{
	public  class VoiceXML : global::haxe.lang.HxObject {
		static VoiceXML() {
			#line 17 "F:\\nx3\\src\\nx3\\xml\\VoiceXML.hx"
			global::nx3.xml.VoiceXML.XVOICE = "voice";
			global::nx3.xml.VoiceXML.XVOICE_TYPE = "type";
			global::nx3.xml.VoiceXML.XVOICE_BARPAUSE = "barpause";
			global::nx3.xml.VoiceXML.XVOICE_DIRECTION = "direction";
		}
		public    VoiceXML(global::haxe.lang.EmptyObject empty){
			unchecked {
				#line 15 "F:\\nx3\\src\\nx3\\xml\\VoiceXML.hx"
				{
				}
				
			}
			#line default
		}
		
		
		public    VoiceXML(){
			unchecked {
				#line 15 "F:\\nx3\\src\\nx3\\xml\\VoiceXML.hx"
				global::nx3.xml.VoiceXML.__hx_ctor_nx3_xml_VoiceXML(this);
			}
			#line default
		}
		
		
		public static   void __hx_ctor_nx3_xml_VoiceXML(global::nx3.xml.VoiceXML __temp_me151){
			unchecked {
				#line 15 "F:\\nx3\\src\\nx3\\xml\\VoiceXML.hx"
				{
				}
				
			}
			#line default
		}
		
		
		public static  string XVOICE;
		
		public static  string XVOICE_TYPE;
		
		public static  string XVOICE_BARPAUSE;
		
		public static  string XVOICE_DIRECTION;
		
		public static   global::Xml toXml(global::nx3.NVoice voice){
			unchecked {
				#line 24 "F:\\nx3\\src\\nx3\\xml\\VoiceXML.hx"
				global::Xml xml = global::Xml.createElement("voice");
				#line 27 "F:\\nx3\\src\\nx3\\xml\\VoiceXML.hx"
				{
					#line 27 "F:\\nx3\\src\\nx3\\xml\\VoiceXML.hx"
					global::nx3.EVoiceType _g = voice.type;
					#line 27 "F:\\nx3\\src\\nx3\\xml\\VoiceXML.hx"
					switch (global::Type.enumIndex(_g)){
						case 1:
						{
							#line 30 "F:\\nx3\\src\\nx3\\xml\\VoiceXML.hx"
							xml.@set("type", global::Std.@string(voice.type));
							#line 30 "F:\\nx3\\src\\nx3\\xml\\VoiceXML.hx"
							break;
						}
						
						
						default:
						{
							#line 27 "F:\\nx3\\src\\nx3\\xml\\VoiceXML.hx"
							{
							}
							
							#line 27 "F:\\nx3\\src\\nx3\\xml\\VoiceXML.hx"
							break;
						}
						
					}
					
				}
				
				#line 34 "F:\\nx3\\src\\nx3\\xml\\VoiceXML.hx"
				if (( voice.direction != global::nx3.EDirectionUAD.Auto )) {
					#line 34 "F:\\nx3\\src\\nx3\\xml\\VoiceXML.hx"
					xml.@set("direction", global::Std.@string(voice.direction));
				}
				
				#line 36 "F:\\nx3\\src\\nx3\\xml\\VoiceXML.hx"
				if (( voice.nnotes != default(global::Array<object>) )) {
					#line 38 "F:\\nx3\\src\\nx3\\xml\\VoiceXML.hx"
					int _g1 = 0;
					#line 38 "F:\\nx3\\src\\nx3\\xml\\VoiceXML.hx"
					global::Array<object> _g11 = voice.nnotes;
					#line 38 "F:\\nx3\\src\\nx3\\xml\\VoiceXML.hx"
					while (( _g1 < _g11.length )){
						#line 38 "F:\\nx3\\src\\nx3\\xml\\VoiceXML.hx"
						global::nx3.NNote note = ((global::nx3.NNote) (_g11[_g1]) );
						#line 38 "F:\\nx3\\src\\nx3\\xml\\VoiceXML.hx"
						 ++ _g1;
						global::Xml noteXml = global::nx3.xml.NoteXML.toXml(note);
						xml.addChild(noteXml);
					}
					
				}
				
				#line 44 "F:\\nx3\\src\\nx3\\xml\\VoiceXML.hx"
				return xml;
			}
			#line default
		}
		
		
		public static   global::nx3.NVoice fromXmlStr(string xmlStr){
			unchecked {
				#line 49 "F:\\nx3\\src\\nx3\\xml\\VoiceXML.hx"
				global::Xml xml = global::Xml.parse(xmlStr).firstElement();
				#line 52 "F:\\nx3\\src\\nx3\\xml\\VoiceXML.hx"
				string typeStr = xml.@get("type");
				global::nx3.EVoiceType type = global::cx.EnumTools.createFromString<global::nx3.EVoiceType>(typeof(global::nx3.EVoiceType), typeStr);
				#line 55 "F:\\nx3\\src\\nx3\\xml\\VoiceXML.hx"
				string directionStr = xml.@get("direction");
				global::nx3.EDirectionUAD direction = default(global::nx3.EDirectionUAD);
				if (string.Equals(directionStr, default(string))) {
					#line 57 "F:\\nx3\\src\\nx3\\xml\\VoiceXML.hx"
					direction = global::nx3.EDirectionUAD.Auto;
				}
				 else {
					#line 57 "F:\\nx3\\src\\nx3\\xml\\VoiceXML.hx"
					direction = global::cx.EnumTools.createFromString<global::nx3.EDirectionUAD>(typeof(global::nx3.EDirectionUAD), directionStr);
				}
				
				#line 60 "F:\\nx3\\src\\nx3\\xml\\VoiceXML.hx"
				global::Array<object> notes = new global::Array<object>(new object[]{});
				#line 62 "F:\\nx3\\src\\nx3\\xml\\VoiceXML.hx"
				{
					#line 62 "F:\\nx3\\src\\nx3\\xml\\VoiceXML.hx"
					object __temp_iterator178 = xml.elements();
					#line 62 "F:\\nx3\\src\\nx3\\xml\\VoiceXML.hx"
					while (global::haxe.lang.Runtime.toBool(global::haxe.lang.Runtime.callField(__temp_iterator178, "hasNext", 407283053, default(global::Array)))){
						#line 62 "F:\\nx3\\src\\nx3\\xml\\VoiceXML.hx"
						global::Xml n = ((global::Xml) (global::haxe.lang.Runtime.callField(__temp_iterator178, "next", 1224901875, default(global::Array))) );
						#line 64 "F:\\nx3\\src\\nx3\\xml\\VoiceXML.hx"
						global::nx3.NNote note = global::nx3.xml.NoteXML.fromXmlStr(n.toString());
						notes.push(note);
					}
					
				}
				
				#line 68 "F:\\nx3\\src\\nx3\\xml\\VoiceXML.hx"
				return new global::nx3.NVoice(((global::Array<object>) (notes) ), ((global::nx3.EVoiceType) (type) ), ((global::nx3.EDirectionUAD) (direction) ));
			}
			#line default
		}
		
		
		public static   bool test(global::nx3.NVoice item){
			unchecked {
				#line 73 "F:\\nx3\\src\\nx3\\xml\\VoiceXML.hx"
				string str = global::nx3.xml.VoiceXML.toXml(item).toString();
				global::nx3.NVoice item2 = global::nx3.xml.VoiceXML.fromXmlStr(str);
				string str2 = global::nx3.xml.VoiceXML.toXml(item2).toString();
				global::haxe.Log.trace.__hx_invoke2_o(default(double), str, default(double), new global::haxe.lang.DynamicObject(new global::Array<int>(new int[]{302979532, 1547539107, 1648581351}), new global::Array<object>(new object[]{"test", "nx3.xml.VoiceXML", "VoiceXML.hx"}), new global::Array<int>(new int[]{1981972957}), new global::Array<double>(new double[]{((double) (76) )})));
				global::haxe.Log.trace.__hx_invoke2_o(default(double), str2, default(double), new global::haxe.lang.DynamicObject(new global::Array<int>(new int[]{302979532, 1547539107, 1648581351}), new global::Array<object>(new object[]{"test", "nx3.xml.VoiceXML", "VoiceXML.hx"}), new global::Array<int>(new int[]{1981972957}), new global::Array<double>(new double[]{((double) (77) )})));
				return string.Equals(str, str2);
			}
			#line default
		}
		
		
		public static  new object __hx_createEmpty(){
			unchecked {
				#line 15 "F:\\nx3\\src\\nx3\\xml\\VoiceXML.hx"
				return new global::nx3.xml.VoiceXML(((global::haxe.lang.EmptyObject) (global::haxe.lang.EmptyObject.EMPTY) ));
			}
			#line default
		}
		
		
		public static  new object __hx_create(global::Array arr){
			unchecked {
				#line 15 "F:\\nx3\\src\\nx3\\xml\\VoiceXML.hx"
				return new global::nx3.xml.VoiceXML();
			}
			#line default
		}
		
		
	}
}

File: Nx3Java/java/src/nx3/xml/VoiceXML.java
package nx3.xml;
import haxe.root.*;

@SuppressWarnings(value={"rawtypes", "unchecked"})
public  class VoiceXML extends haxe.lang.HxObject
{
	static 
	{
		nx3.xml.VoiceXML.XVOICE = "voice";
		nx3.xml.VoiceXML.XVOICE_TYPE = "type";
		nx3.xml.VoiceXML.XVOICE_BARPAUSE = "barpause";
		nx3.xml.VoiceXML.XVOICE_DIRECTION = "direction";
	}
	public    VoiceXML(haxe.lang.EmptyObject empty)
	{
		{
		}
		
	}
	
	
	public    VoiceXML()
	{
		nx3.xml.VoiceXML.__hx_ctor_nx3_xml_VoiceXML(this);
	}
	
	
	public static   void __hx_ctor_nx3_xml_VoiceXML(nx3.xml.VoiceXML __temp_me195)
	{
		{
		}
		
	}
	
	
	public static  java.lang.String XVOICE;
	
	public static  java.lang.String XVOICE_TYPE;
	
	public static  java.lang.String XVOICE_BARPAUSE;
	
	public static  java.lang.String XVOICE_DIRECTION;
	
	public static   haxe.root.Xml toXml(nx3.NVoice voice)
	{
		haxe.root.Xml xml = haxe.root.Xml.createElement("voice");
		{
			nx3.EVoiceType _g = voice.type;
			switch (haxe.root.Type.enumIndex(_g))
			{
				case 1:
				{
					xml.set("type", haxe.root.Std.string(voice.type));
					break;
				}
				
				
				default:
				{
					{
					}
					
					break;
				}
				
			}
			
		}
		
		if (( voice.direction != nx3.EDirectionUAD.Auto )) 
		{
			xml.set("direction", haxe.root.Std.string(voice.direction));
		}
		
		if (( voice.nnotes != null )) 
		{
			int _g1 = 0;
			haxe.root.Array<nx3.NNote> _g11 = voice.nnotes;
			while (( _g1 < _g11.length ))
			{
				nx3.NNote note = _g11.__get(_g1);
				 ++ _g1;
				haxe.root.Xml noteXml = nx3.xml.NoteXML.toXml(note);
				xml.addChild(noteXml);
			}
			
		}
		
		return xml;
	}
	
	
	public static   nx3.NVoice fromXmlStr(java.lang.String xmlStr)
	{
		haxe.root.Xml xml = haxe.root.Xml.parse(xmlStr).firstElement();
		java.lang.String typeStr = xml.get("type");
		nx3.EVoiceType type = cx.EnumTools.createFromString(nx3.EVoiceType.class, typeStr);
		java.lang.String directionStr = xml.get("direction");
		nx3.EDirectionUAD direction = null;
		if (( directionStr == null )) 
		{
			direction = nx3.EDirectionUAD.Auto;
		}
		 else 
		{
			direction = cx.EnumTools.createFromString(nx3.EDirectionUAD.class, directionStr);
		}
		
		haxe.root.Array<nx3.NNote> notes = new haxe.root.Array<nx3.NNote>(new nx3.NNote[]{});
		{
			java.lang.Object __temp_iterator249 = xml.elements();
			while (haxe.lang.Runtime.toBool(haxe.lang.Runtime.callField(__temp_iterator249, "hasNext", null)))
			{
				haxe.root.Xml n = ((haxe.root.Xml) (haxe.lang.Runtime.callField(__temp_iterator249, "next", null)) );
				nx3.NNote note = nx3.xml.NoteXML.fromXmlStr(n.toString());
				notes.push(note);
			}
			
		}
		
		return new nx3.NVoice(((haxe.root.Array<nx3.NNote>) (notes) ), ((nx3.EVoiceType) (type) ), ((nx3.EDirectionUAD) (direction) ));
	}
	
	
	public static   boolean test(nx3.NVoice item)
	{
		java.lang.String str = nx3.xml.VoiceXML.toXml(item).toString();
		nx3.NVoice item2 = nx3.xml.VoiceXML.fromXmlStr(str);
		java.lang.String str2 = nx3.xml.VoiceXML.toXml(item2).toString();
		haxe.Log.trace.__hx_invoke2_o(0.0, str, 0.0, new haxe.lang.DynamicObject(new haxe.root.Array<java.lang.String>(new java.lang.String[]{"className", "fileName", "methodName"}), new haxe.root.Array<java.lang.Object>(new java.lang.Object[]{"nx3.xml.VoiceXML", "VoiceXML.hx", "test"}), new haxe.root.Array<java.lang.String>(new java.lang.String[]{"lineNumber"}), new haxe.root.Array<java.lang.Object>(new java.lang.Object[]{((java.lang.Object) (((double) (76) )) )})));
		haxe.Log.trace.__hx_invoke2_o(0.0, str2, 0.0, new haxe.lang.DynamicObject(new haxe.root.Array<java.lang.String>(new java.lang.String[]{"className", "fileName", "methodName"}), new haxe.root.Array<java.lang.Object>(new java.lang.Object[]{"nx3.xml.VoiceXML", "VoiceXML.hx", "test"}), new haxe.root.Array<java.lang.String>(new java.lang.String[]{"lineNumber"}), new haxe.root.Array<java.lang.Object>(new java.lang.Object[]{((java.lang.Object) (((double) (77) )) )})));
		return haxe.lang.Runtime.valEq(str, str2);
	}
	
	
	public static   java.lang.Object __hx_createEmpty()
	{
		return new nx3.xml.VoiceXML(((haxe.lang.EmptyObject) (haxe.lang.EmptyObject.EMPTY) ));
	}
	
	
	public static   java.lang.Object __hx_create(haxe.root.Array arr)
	{
		return new nx3.xml.VoiceXML();
	}
	
	
}


